### PR TITLE
fix: update dojo to v1.17.3 [PPP-4824]

### PR DESCRIPTION
**Pull request by davidmsantos90**
Update from dojo **v1.9.2** to **v1.17.3**

@pentaho/millenniumfalcon please review

### Overrides
| File | Source update in 1.17 | Keep Override |  description  |
| :--- | :--------------------: | :-------------: | --- |
| dijit/_CssStateMixin.js | ❌ | ✅ |  | 
| dijit/_HasDropDown.js | ✅ | ✅ |  | 
| dijit/_PaletteMixin.js | ❌ | ✅ |  | 
| dijit/Dialog.js | ✅ | ✅ |  | 
| dijit/Menu.js | ✅ | ✅ |  | 
|  |  |  |
| dojo/_base/config.js | ✅ | ✅ |  |
| dojo/_base/kernel.js | ✅ | ❌ | fixed: global properly defined |
| dojo/request/default.js | ✅ | ✅ |  |
| dojo/selector/_loader.js | ✅ | ❌ | fixed: r.js build |
| dojo/store/JsonRest.js | ✅ | ❌ | override was file from v1.17.3 |
| dojo/store/Memory.js | ✅ | ❌ | fixed: add/put methods use `options` arg |
| dojo/dom-geometry.js | ✅ | ❌ | fixed: return empty rectangle on error |
| dojo/dom-prop.js | ✅ | ✅ |  |
| dojo/i18n.js | ✅ | ✅ |  |
| dojo/on.js | ✅ | ❌ | override was from v1.9.6 |
|  |  |  |
| dojox/grid/_FocusManager.js | ❌ | ❌ | only override was `array.some` instead of `dojo.some` |
| dojox/grid/_Scroller.js | ❌ | ❌ | only override was `array.forEach` instead of `dojo.forEach` |
| dojox/grid/_View.js | ✅ | ✅ | remove "dojo" require |
| dojox/grid/Selection.js | ❌ | ✅ | force ctrl as copy key |
| dojox/json/ref.js | ✅ | ✅ | remove "dojox" require |
| dojox/json/schema.js | ❌ | ✅ | remove "dojox" require |
| dojox/layout/ReziseHandle.js | ✅ | ❌ | only override was `declare` instead of `dojo.declare` |
| dojox/rpc/JsonRest.js | ❌ | ✅ |  |
| dojox/rpc/Rest.js | ❌ | ✅ |  |
| dojox/storage/_common.js | ❌ | ✅ | original source is not AMD |
| dojox/storage/LocalStorageProvider.js | ✅ | ❌ | is AMD in new version, override not required |
| dojox/storage/manager.js | ✅ | ❌ | is AMD in new version, override not required |
| dojox/storage/Provider.js | ✅ | ❌ | is AMD in new version, override not required |
| dojox/storage/WhatWGStorageProvider.js | ❌ | ❌ | override is equal to original source |
| dojox/xml/parser.js | ❌ | ❌ | just defining dojox if not defined |
| dojox/storage.js | ❌ | ❌ | override is equal to original source |

### CODE REVIEW

**Comment by buildguy** [comment]
<div align='center'>

[👍 Frogbot scanned this pull request and found that it did not add vulnerable dependencies.](https://docs.jfrog-applications.jfrog.io/jfrog-applications/frogbot)

</div>



---
<div align='center'>

[🐸 JFrog Frogbot](https://docs.jfrog-applications.jfrog.io/jfrog-applications/frogbot)

</div>

**Comment by buildguy** [comment]
### :white_check_mark: Build finished in 12m 11s
Build command:
```console
mvn clean verify -B -e -Daudit -Djs.no.sandbox
```
---
:ok_hand: All tests passed!
 
Tests run: **5198**, Failures: **0**, Skipped: **5** &nbsp;&nbsp; [Test Results](https://wingman.orl.eng.hitachivantara.com/job/projects/job/common-ui/job/PR-1788/1/testReport)

 ---
 
 

:information_source:  _This is an automatic message_

**Comment by buildguy** [comment]
### :x: Build failed in 40s
Build command:
```console
mvn clean verify -B -e -Daudit -Djs.no.sandbox
```
---
 :exclamation: No tests found!
 ---
### Errors:
<details><summary><b>Filtered log</b> (click to expand)</summary>
<p>

```
script returned exit code 1
```
 
</p>
</details>
 
 

:information_source:  _This is an automatic message_

**Review by mbrasil — APPROVED** [review]

**Comment by buildguy** [comment]
<div align='center'>

[👍 Frogbot scanned this pull request and found that it did not add vulnerable dependencies.](https://docs.jfrog-applications.jfrog.io/jfrog-applications/frogbot)

</div>



---
<div align='center'>

[🐸 JFrog Frogbot](https://docs.jfrog-applications.jfrog.io/jfrog-applications/frogbot)

</div>

**Comment by buildguy** [comment]
### :white_check_mark: Build finished in 13m 25s
Build command:
```console
mvn clean verify -B -e -Daudit -Djs.no.sandbox
```
---
:ok_hand: All tests passed!
 
Tests run: **5198**, Failures: **0**, Skipped: **5** &nbsp;&nbsp; [Test Results](https://wingman.orl.eng.hitachivantara.com/job/projects/job/common-ui/job/PR-1788/3/testReport)

 ---
 
 

:information_source:  _This is an automatic message_

File: pom.xml [hunk 1]
```diff
@@ -45,11 +44,26 @@
...
+        <artifactId>dojox</artifactId>
+        <version>${dojo.version}</version>
+      </dependency>
+      <dependency>
+        <groupId>org.webjars.npm</groupId>
+        <artifactId>dijit</artifactId>
+        <version>${dojo.version}</version>
+      </dependency>
+      <dependency>
+        <groupId>org.dojotoolkit</groupId>
```
Comment (dcleao) [inline pom.xml:62]: Still needed?

**Review by dcleao — CHANGES_REQUESTED** [review]

File: pom.xml [hunk 2]
```diff
@@ -29,7 +29,6 @@
   </scm>
 
   <properties>
-    <dojo.version>1.9.2</dojo.version>
```
Comment (dcleao) [inline pom.xml:32]: Nice improvement, moving to parent pom!

**Review by dcleao — APPROVED** [review]
💯 💯 !

**Comment by buildguy** [comment]
<div align='center'>

[👍 Frogbot scanned this pull request and found that it did not add vulnerable dependencies.](https://docs.jfrog-applications.jfrog.io/jfrog-applications/frogbot)

</div>



---
<div align='center'>

[🐸 JFrog Frogbot](https://docs.jfrog-applications.jfrog.io/jfrog-applications/frogbot)

</div>

**Pull request merged**